starport_template/lib/utils/cosmos_balances.dart

```diff
@@ -0,0 +1,40 @@
+import 'package:flutter/foundation.dart';
+import 'package:starport_template/entities/amount.dart';
+import 'package:starport_template/entities/balance.dart';
+import 'package:starport_template/entities/denom.dart';
+import 'package:starport_template/model/balance_json.dart';
+import 'package:starport_template/starport_app.dart';
+import 'package:starport_template/utils/base_env.dart';
+import 'package:http/http.dart' as http;
+
+class CosmosBalances {
+  BaseEnv baseEnv;
+
+  CosmosBalances(this.baseEnv);
+
+  Future<void> getCosmosBalances(BaseEnv baseEnv, String walletAddress) async {
+    StarportApp.walletsStore.isError.value = false;
```

Comment:
we're mixing too many things here. I just wanted to extract the logic for getting balances away from WalletStore. here, we're manipulating store's internals outside of the store class and as well making the `CosmosBalances` class be dependent on the `WalletStore` which tightly couples those two classes in an inverted way.

We can have `getBalances` method inside the `WalletStore`, but instead of having the whole logic there, it should simply call `CosmosBalances.getBalances()` internally, something like this:

```dart
class WalletStore {
...
Future<void>  getBalances(String walletAddress) {
  isError.value = false;
  isBalancesLoading.value = true;
  try {
    balancesList.value = await CosmosBalances(baseEnv).getBalances();
  } catch (error) {
    isError.value = false;
  }
  isBalancesLoading.value = false;
}

Reply:
✅